pyomo/contrib/mindtpy/iterate.py

```diff
@@ -172,6 +179,24 @@ def algorithm_should_terminate(solve_data, config, check_cycling):
         solve_data.results.solver.termination_condition = tc.maxTimeLimit
         return True
 
+    if config.strategy == 'ECP':
+        MindtPy = solve_data.working_model.MindtPy_utils
+        nonlinear_constraints = [c for c in MindtPy.constraint_list if
+                                 c.body.polynomial_degree() not in (1, 0)]
+        for nlc in nonlinear_constraints:
+            if nlc.has_lb():
+                if nlc.lslack() < -config.ECP_tolerance:
+                    return False
+            if nlc.has_ub():
+                if nlc.uslack() < -config.ECP_tolerance:
+                    return False
```

Comment:
Before the return could you have the logger reporting which constraint is still violated?